applications/test_game/systems/testsystem.hpp

```diff
@@ -52,23 +54,36 @@ class TestSystem final : public System<TestSystem>
         app::InputSystem::createBinding<player_fly>(app::inputmap::method::LEFT_SHIFT, -1.f);
         app::InputSystem::createBinding<player_look_x>(app::inputmap::method::MOUSE_X, 1.f);
         app::InputSystem::createBinding<player_look_y>(app::inputmap::method::MOUSE_Y, 1.f);
+        app::InputSystem::createBinding<exit_action>(app::inputmap::method::ESCAPE);
 
         bindToEvent<player_move, &TestSystem::onPlayerMove>();
         bindToEvent<player_strive, &TestSystem::onPlayerStrive>();
         bindToEvent<player_fly, &TestSystem::onPlayerFly>();
         bindToEvent<player_look_x, &TestSystem::onPlayerLookX>();
         bindToEvent<player_look_y, &TestSystem::onPlayerLookY>();
+        bindToEvent<exit_action, &TestSystem::onExit>();
+
+        app::window window = m_ecs->world.get_component_handle<app::window>().read();
+        window.enableCursor(false);
+        rendering::model_handle modelH;
+        rendering::material_handle wireframeH;
+        rendering::material_handle vertexH;
+
+        {
+            async::readwrite_guard guard(*window.lock);
+            app::ContextHelper::makeContextCurrent(window);
 
-        bindToEvent<events::component_creation<app::window>, &TestSystem::disableCursor>();
+            modelH = rendering::ModelCache::create_model("test", "basic://models/Cube.obj"_view);
+            wireframeH = rendering::MaterialCache::create_material("wireframe", "basic:/shaders/wireframe.glsl"_view);
+            vertexH = rendering::MaterialCache::create_material("vertex", "basic:/shaders/position.glsl"_view);
```

Comment:
`basic:/`vs `basic://` ?

Reply:
whoops accidental typo, it works tho

Reply:
I'll cahnge it before merge

Reply:
yes I made the api, specifically look out for that error.
It is a "hidden" feature xD 
https://github.com/Args-Engine/Args-Engine/blob/main/args/core/filesystem/navigator.cpp
```cpp
 	//find root domain
    // the syntax for inline root-domains has to be <root-domain>:[/\\]+<rest-of-the-path>
    const auto rdIndex = m_path.find_first_of(':');
    if(rdIndex == std::string::npos) return Err(args_fs_error("invalid syntax for path string, no domain delimiter"));

    	root_domain = m_path.substr(0,rdIndex);

        //find the rest of the path ... technically this will resolve //////////lol/test.bin to lol/test.bin just fine,
		//but hopefully no one is actually insane enough to mess with such paths
		//
		//if he/she does, congrats you found a ... "feature"
        const auto pIndex = m_path.find_first_not_of("/\\",rdIndex+1);
        if(pIndex == std::string::npos) return Err(args_fs_error("invalid sytax for path string, last domain delimiter not found"));
          
        to_process = m_p
```